hazelcast/src/main/java/com/hazelcast/cp/internal/operation/RaftReplicateOp.java

```diff
@@ -62,6 +63,8 @@ public final void run() {
         if (raftNode == null) {
             if (service.isRaftGroupDestroyed(groupId)) {
                 sendResponse(new CPGroupDestroyedException(groupId));
+            } else if (isHeartbeatSessionOp()) {
```

Comment:
instead of pulling operation-specific functionality in `RaftReplicateOp` infrastructure, can we deal with it in a different way? Maybe eg add `void onNullRaftNode()` method to `RaftOp` to allow operations to customize how they respond to raft node being `null` here?

Reply:
Thanks Vassilis, I'll see how to make the code clear.